crates/core_arch/build.rs

```diff
@@ -1,3 +1,372 @@
+const INT_TYPES: [&'static str; 8] = [
+    "int8x8_t",
+    "int8x16_t",
+    "int16x4_t",
+    "int16x8_t",
+    "int32x2_t",
+    "int32x4_t",
+    "int64x1_t",
+    "int64x2_t",
+];
+
+/*
+const FLOAT_TYPES: [&'static str; 6] = [
+    "float16x4_t",
+    "float16x8_t",
+    "float32x2_t",
+    "float32x4_t",
+    "float64x1_t",
+    "float64x2_t",
+];
+*/
+
+fn type_len(t: &str) -> usize {
+    match t {
+        "int8x8_t" => 8,
+        "int8x16_t" => 16,
+        "int16x4_t" => 4,
+        "int16x8_t" => 8,
+        "int32x2_t" => 2,
+        "int32x4_t" => 4,
+        "int64x1_t" => 1,
+        "int64x2_t" => 2,
+        "uint8x8_t" => 8,
+        "uint8x16_t" => 16,
+        "uint16x4_t" => 4,
+        "uint16x8_t" => 8,
+        "uint32x2_t" => 2,
+        "uint32x4_t" => 4,
+        "uint64x1_t" => 1,
+        "uint64x2_t" => 2,
+        "float16x4_t" => 4,
+        "float16x8_t" => 8,
+        "float32x2_t" => 2,
+        "float32x4_t" => 4,
+        "float64x1_t" => 1,
+        "float64x2_t" => 2,
+        "poly64x1_t" => 1,
+        "poly64x2_t" => 2,
+        _ => panic!("unknown type: {}", t),
+    }
+}
+
+fn type_to_suffix(t: &str) -> &str {
+    match t {
+        "int8x8_t" => "_s8",
+        "int8x16_t" => "q_s8",
+        "int16x4_t" => "_s16",
+        "int16x8_t" => "q_s16",
+        "int32x2_t" => "_s32",
+        "int32x4_t" => "q_s32",
+        "int64x1_t" => "_s64",
+        "int64x2_t" => "q_s64",
+        "uint8x8_t" => "_u8",
+        "uint8x16_t" => "q_u8",
+        "uint16x4_t" => "_u16",
+        "uint16x8_t" => "q_u16",
+        "uint32x2_t" => "_u32",
+        "uint32x4_t" => "q_u32",
+        "uint64x1_t" => "_u64",
+        "uint64x2_t" => "q_u64",
+        "float16x4_t" => "_f16",
+        "float16x8_t" => "q_f16",
+        "float32x2_t" => "_f32",
+        "float32x4_t" => "q_f32",
+        "float64x1_t" => "_f64",
+        "float64x2_t" => "q_f64",
+        "poly64x1_t" => "_p64",
+        "poly64x2_t" => "q_p64",
+        _ => panic!("unknown type: {}", t),
+    }
+}
+
+fn type_to_global_type(t: &str) -> &str {
+    match t {
+        "int8x8_t" => "i8x8",
+        "int8x16_t" => "i8x16",
+        "int16x4_t" => "i16x4",
+        "int16x8_t" => "i16x8",
+        "int32x2_t" => "i32x2",
+        "int32x4_t" => "i32x4",
+        "int64x1_t" => "i64x1",
+        "int64x2_t" => "i64x2",
+        "uint8x8_t" => "u8x8",
+        "uint8x16_t" => "u8x16",
+        "uint16x4_t" => "u16x4",
+        "uint16x8_t" => "u16x8",
+        "uint32x2_t" => "u32x2",
+        "uint32x4_t" => "u32x4",
+        "uint64x1_t" => "u64x1",
+        "uint64x2_t" => "u64x2",
+        "float16x4_t" => "f16x4",
+        "float16x8_t" => "f16x8",
+        "float32x2_t" => "f32x2",
+        "float32x4_t" => "f32x4",
+        "float64x1_t" => "f64",
+        "float64x2_t" => "f64x2",
+        "poly64x1_t" => "i64x1",
+        "poly64x2_t" => "i64x2",
+        _ => panic!("unknown type: {}", t),
+    }
+}
+
+fn main() -> io::Result<()> {
```

Comment:
All this code always runs, but it should only run when explicitly requested.

Reply:
That's intentional, running always (with no effect on no change) enforces generated files to be checked in so people couldn't accidentally forget to commit changes.

Reply:
Moved this to an example so it needs to be manually ran & committed so it's only affecting people purposefully adding intrinsics.